### DISCUSSION TITLE
Triple buffering patch in muffin?

### DISCUSSION THREAD

**Discussion by sertraline** [linuxmint/discussions#506] [Questions]
There is a triple buffering [patch for mutter](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1441). While it hasn't been merged into upstream yet, it has been 3 years since it's out. Ubuntu and Debian already include this patch with the GNOME package, and generally compositing experience is way smoother on their side. Are there any plans to integrate that patch into muffin?

**Comment by Jazzhoo** [comment]
Triple buffer patch has been merged and is planned for Gnome 48. Whether we ever see it ported to Cinnamon/Muffin is a good question.

**Comment by naguam** [comment]
I second the question.

Linux mint is renown to be used to give second life to old hardware.

Cinnamon (Especially the menu) stutters on my Latitude e4310 with i5 560M and Ironlake Mobile GPU.

Triple buffering even makes gnome a smoother option (Still some stutter because gnome animations are heavy and that igpu is stuck to OpenGL 2.1).

**Reply by roxen-creator** [reply]
I completely agree with @naguam

I have an accer nitro 5 with Intel© Core™ i5-10300H CPU and Intel Corporation CometLake-H GT2 [UHD Graphics] with dedicated GTX 1650. I use nvidia on demand on nvidia prime.

I have also been struggling with the cinammon window effects the animation are very laggy it is a visual only no effect on the performance itself though. But I have to completely remove all the animation to use mint.

However I would love see any improvement in the cinammon de that would enhance animation effects and make it smooth.
